src/clj/fluree/db/flake/reasoner.cljc

```diff
@@ -163,28 +165,47 @@
     (log/debug "Reasoner - source OWL rules: " graph)
     (owl-datalog/owl->datalog inserts graph)))
 
+(defn rules-from-dbs
+  [methods inserts dbs]
+  (go-try
+    (loop [[method & remaining-methods] methods
+           rules []]
+      (if method
+        (recur remaining-methods
+               (loop [[db & remaining-dbs] dbs
+                      rules* rules]
+                 (if db
+                   (recur remaining-dbs
+                          (into rules*
+                                (as-> db $
+                                  (<? (resolve/rules-from-db $ method))
+                                  (rules-from-graph method inserts $))))
+                   rules*)))
+        (remove empty? rules)))))
+
 (defn all-rules
   "Gets all relevant rules for the specified methods from the
   supplied rules graph or from the db if no graph is supplied."
-  [methods db inserts graph-or-db]
+  [methods db inserts rule-sources]
   (go-try
-    (let [rules-db        (cond
-                            (nil? graph-or-db) db
-                            (queryable? graph-or-db) graph-or-db)
-          supplied-rules* (when-not rules-db
-                            (try*
-                              (parse-rules-graph graph-or-db)
-                              (catch* e
-                                      (log/error "Error parsing supplied rules graph:" e)
-                                      (throw e))))]
-      (loop [[method & r] methods
-             rules []]
-        (if method
-          (let [rules-graph* (or supplied-rules*
-                                 (<? (resolve/rules-from-db rules-db method)))
-                rules*       (rules-from-graph method inserts rules-graph*)]
-            (recur r (into rules rules*)))
-          rules)))))
+    (let [rule-graphs           (filter #(and (map? %) (not (queryable? %))) rule-sources)
```

Comment:
It looks like you have `fluree.db.query.exec` required twice. I prefer not `:refer`ing functions so it's easy to tell at a glance what dependencies a function has just by looking at the ns, so I think just using `exec/queryable` would be fine here. About the only thing I ever `:refer` is macros like `go-try` and `try*`/`catch*`.